libafl/src/events/centralized.rs

```diff
@@ -799,10 +800,11 @@ where
         event: Event<<EM::State as UsesInput>::Input>,
     ) -> Result<(), Error>
     where
-        E: Executor<Self, Z> + HasObservers<State = EM::State>,
+        E: Executor<Self, Z> + HasObservers<State = <Self as UsesState>::State>,
         EM::State: UsesInput + HasExecutions + HasMetadata,
```

Comment:
Should technically also be `Self::State` then?

Reply:
Yes, but in some part you can't use Self::State. because it complains. "ambigious something.." error